server/utils/passport/local-login.js

```diff
@@ -19,64 +20,47 @@ export default (req, res, next, user) => {
     done(null, user)
   })
   //If the user exists, serialize the user to the session
-  req.login(user, function (err) {
-    const { appDb } = req.app.locals
-    const uid = user.uid
+  req.login(user, async function (err) {
+    if (err) return res.redirect(`${routes.login}?e=${err}`)
 
-    appDb.collection(collections.configs).findOne(
-      {
-        owner: uid,
-      },
-      function (err, configObj) {
-        if (err) {
-          console.error(err.message)
-        }
-        if (!configObj) {
-          const defaultConfig = {
-            owner: uid,
-            config: defaultUserConfig,
-            name: 'default',
-            type: 'matrix',
-            readers: [uid],
-            created: new Date().toUTCString(),
-          }
-          appDb.collection(collections.configs).insertOne(defaultConfig)
-        }
-        appDb.collection(collections.users).findOneAndUpdate(
-          { uid: user.uid },
-          {
-            $set: {
-              last_logon: Date.now(),
-            },
-          },
-          {
-            projection: {
-              _id: 0,
-              uid: 1,
-              display_name: 1,
-              acl: 1,
-              role: 1,
-              icon: 1,
-              mail: 1,
-              access: 1,
-            },
-            returnOriginal: false,
-            upsert: true,
-          },
-          function (err, userinfo) {
-            if (err) {
-              console.error(err.message)
-            }
-            req.session.role = userinfo.value.role
-            req.session.display_name = userinfo.value.display_name
-            req.session.mail = userinfo.value.mail
-            req.session.celery_tasks = []
-            req.session.icon = userinfo.value.icon
-            req.session.userAccess = userinfo.value.access
-            return res.redirect(`${basePath}/`)
-          }
-        )
+    try {
+      const { appDb } = req.app.locals
+      const uid = user.uid
+      const config = await ConfigModel.find(appDb, uid)
+
+      if (!config) {
+        const configAttributes = { owner: uid, readers: [uid] }
+
+        await ConfigModel.save(appDb, configAttributes)
       }
-    )
+
+      const userInfo = await UserModel.findAndUpdate(
+        appDb,
+
+        uid,
+
+        {
+          last_logon: Date.now(),
+        }
+      )
```

Comment:
nit: we can condense the lines here.

```suggestion
      const userInfo = await UserModel.findAndUpdate(
        appDb,
        uid,
        { last_logon: Date.now() },
      )
```